src/game/systems/ItemInteractionSystem.ts

```diff
@@ -0,0 +1,462 @@
+import { createEntitiesFromTemplates } from '../utils/EntityFactory';
+import { PositionComponent } from '../components/individualComponents/PositionComponent';
+
+/**
+ * ItemInteractionSystem handles interactions between entities that require items
+ * and entities that provide usable items. It processes capability matching,
+ * item consumption, and interaction behaviors.
+ */
+export class ItemInteractionSystem implements System {
+  /**
+   * Updates the item interaction system, processing all active interactions
+   * between entities that require items and entities that provide items.
+   *
+   * @param args - Update arguments containing entities array
+   */
+  update(args: UpdateArgs): void {
+    const { entities } = args;
+
+    // Process all current interactions
+    this.processInteractions(entities);
+  }
+
+  /**
+   * Processes all active interactions between entities.
+   * Finds entities that are currently interacting and validates their capabilities.
+   *
+   * @param entities - Array of all entities to process
+   * @private
+   */
+  private processInteractions(entities: Entity[]): void {
+    // Get all entities that are currently in an interacting state
+    const interactingEntities = getEntitiesWithComponents(
+      [ComponentType.Interacting],
+      entities,
+    );
+
+    for (const entity of interactingEntities) {
+      // Find entities that require items and are currently interacting
+      const requiresItemEntities = this.findRequiredItemEntities(
+        entities,
+        entity,
+      );
+
+      // Process each requirement
+      for (const requiredEntity of requiresItemEntities) {
+        this.validateCapabilities(entity, requiredEntity, entities);
+      }
+    }
+  }
+
+  /**
+   * Finds entities that require items and are at the same position as the interacting entity.
+   * Uses position-based proximity logic to determine valid interaction targets.
+   *
+   * @param entities - Array of all entities
+   * @param interactingEntity - The entity currently interacting
+   * @returns Array of entities that require items for interaction at the same position
+   * @private
+   */
+  private findRequiredItemEntities(
+    entities: Entity[],
+    interactingEntity: Entity,
+  ): Entity[] {
+    // Get the position of the interacting entity
+    const interactingPosition = getComponentIfExists(
+      interactingEntity,
+      ComponentType.Position,
+    ) as PositionComponent;
+
+    if (!interactingPosition) {
+      console.log('Interacting entity does not have Position component');
+      return [];
+    }
+
+    // Find entities with RequiresItem component that are at the same position and active
+    return getEntitiesWithComponents(
+      [ComponentType.RequiresItem],
+      entities,
+    ).filter((entity) => {
+      const requiresComponent = getComponentIfExists(
+        entity,
+        ComponentType.RequiresItem,
+      ) as RequiresItemComponent;
```

Comment:
I think this can be typed. Prefer this pattern everywhere to the ` as ` pattern.
```suggestion
      const requiresComponent = getComponentIfExists<RequiresItemComponent>(
        entity,
        ComponentType.RequiresItem,
      );
```